aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -106,14 +108,20 @@ module aptos_framework::object {
         /// transfers bypassing the use of a TransferRef.
         allow_ungated_transfer: bool,
         /// Emitted events upon transferring of ownership.
-        transfer_events: event::EventHandle<TransferEvent>,
+        transfer_events: event::EventHandle<TransferEvent>
     }
 
     #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
     /// This is added to objects that are burnt (ownership transferred to BURN_ADDRESS).
     struct TombStone has key {
         /// Track the previous owner before the object is burnt so they can reclaim later if so desired.
-        original_owner: address,
+        original_owner: address
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// This is used to prevent the object from being deleted if the counter is not 0.
+    struct DeletionGuard has key {
+        counter: u16
```

Comment:
This doesn't really protect except against people who add it this way?  Maybe it's better to make something else that can check how many resources are in the ObjectGroup?